### PR TITLE
Ensure list view uses FH results

### LINKED ISSUE #63: Filter list of results to only FH patients

### PR DESCRIPTION
**Pull request by stefpiatek**
We're now implementing the results list for FH patients and using the new structure of the FHIR bundle 🎉 

closes #63 

- Results lists show FH reports only
- Results lists shows overall interpretation
- Specific modal for (and tests):
  - positive result and other family members tested
  - positive result and no other famliy members tested
  - negative result
  - postitive result and no family number for patient
-  Also rationalised the setup of tests so that we're not repeating FHIR requests that aren't needed 
- Ensured loading spinner is shown while results are fetched for list view

### CODE REVIEW

File: src/components/results-list/ResultsDataFetcher.tsx [hunk 1]
```diff
@@ -141,16 +152,20 @@ const ResultsDataFetcher: FC = () => {
     return readableResults;
   };
 
-  if (!parsedResults) {
-    return <div>Something went wrong getting observations. Please try again later.</div>;
+  if (!parsedResults && !isLoading) {
+    return <div>Familial hypercholesterolemia patient results were found on the FHIR server</div>;
```
Comment (stefpiatek) [inline src/components/results-list/ResultsDataFetcher.tsx:156]:
```suggestion
    return <div>No familial hypercholesterolemia patient results were found on the FHIR server</div>;
```

File: src/components/reports/ReportForm.test.tsx [hunk 2]
```diff
@@ -165,6 +165,17 @@ const renderTestReportForm = () => {
...
+  /*
+   * For some reason on first load of FHIR server, creating a batch of requests doesn't seem to validate
+   * Sending a resource before any test are run fixed the issue
+   */
+  const practitioner = new Practitioner();
+  practitioner.resourceType = "Practitioner";
+  practitioner.identifier = [createIdentifier("initial")];
+  return createPractitioner(practitioner);
+});
+
```
Comment (jhughes982) [inline src/components/reports/ReportForm.test.tsx:178]: Interesting, is this a frontend issue I wonder? Does it work if you interact with the backend directly? Hopefully this doesn't cause any user interaction issues on the deployed application.

**Review by jhughes982 — APPROVED** [review]
Looks good 👍

File: src/components/reports/ReportForm.test.tsx (hunk 2 above)
Comment (stefpiatek) [reply src/components/reports/ReportForm.test.tsx:178]: yeah it's only the first time a FHIR server is spun up that it happens which is interesting.

**Pull request merged**